### PR TITLE
feat: retest [EOA-6664]

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/final-api/v1/http/ddtf_build.rb [hunk 1]
```diff
@@ -70,13 +70,12 @@ def test_data
             'stoppedBy': build.stopped_by.try(:name), # TODO
 
             'isTsd': true,
-            'checkpoints':    config[:checkpoints],
-            'debugging':      config[:debbuging],
-            'buildSignal':    config[:build_signal],
-            'scenarioScript': config[:scenario_script],
-            'packageSource':  config[:package_source],
+            'checkpoints':    checkpoints,
+            'debugging':      config[:debbuging] || false,
```
Comment (ondrej-hosak) [inline lib/final-api/v1/http/ddtf_build.rb]: debugging

File: lib/final-api/helpers/ddtf_helpers.rb [hunk 2]
```diff
@@ -8,23 +8,24 @@ class DdtfHelpers
...
               language: 'tsd',
               git:
               {
                 no_clone: true
               },
-              name: tsd['name'],
+              name: enqueue_data.tsd['name'],
               description: test_data.get_ikey('Description'),
-              branch: test_data.get_ikey('Package'),
+              branch: enqueue_data.package_source,
```
Comment (ondrej-hosak) [inline lib/final-api/helpers/ddtf_helpers.rb:22]: test_data.get_ikey('Package') || enqueue_data.package_source

File: lib/final-api/v1/http/ddtf_build.rb (hunk 1 above)
Comment (petrSchreiber) [inline lib/final-api/v1/http/ddtf_build.rb]: I remember Zdeněk Novotný already fixed it as part of another issue, so I am surprised it is still here.

Idea - fixing typo would be nice, but I wonder - what is this flag used for and do we still need it?
I went through AVG.sln and did not find any proof of usability of this :)

File: lib/final-api/endpoint/ddtf.rb [hunk 3]
```diff
@@ -39,6 +39,15 @@ def registered(app)
             FinalAPI::V1::Http::DDTF_Build.new(build, {}).test_data.to_json
           end
 
+          app.post '/ddtf/tests/:id/retest' do
+            build = Build.find(params[:id])
+            retest_data = FinalAPI::V1::Http::DDTF_Build.new(build, {}).retest_data
+            retest_data['runtimeConfigFields'].reject! do |key, _|
+              key[:definition].downcase.start_with?("webserver")
```
Comment (petrSchreiber) [inline lib/final-api/endpoint/ddtf.rb]: who trims webservers from the request?

File: lib/final-api/endpoint/ddtf.rb [hunk 4]
```diff
@@ -39,6 +39,15 @@ def registered(app)
             FinalAPI::V1::Http::DDTF_Build.new(build, {}).test_data.to_json
           end
 
+          app.post '/ddtf/tests/:id/retest' do
+            build = Build.find(params[:id])
+            retest_data = FinalAPI::V1::Http::DDTF_Build.new(build, {}).retest_data
+            retest_data['runtimeConfigFields'].reject! do |key, _|
```
Comment (petrSchreiber) [inline lib/final-api/endpoint/ddtf.rb:45]: The delete_if could be a bit more readable, but that is just a question of personal taste and not a merge blocker :)

**Pull request merged**